clients/src/main/java/org/apache/kafka/common/record/Records.java

```diff
@@ -60,32 +60,34 @@
      * record data (see {@link FileLogInputStream.FileChannelLogEntry#magic()}.
      * @return An iterator over the shallow entries of the log
      */
-    Iterable<? extends LogEntry> shallowEntries();
-
-    /**
-     * Get the deep log entries (i.e. descend into compressed message sets). For the deep records,
-     * there are fewer options for optimization since the data must be decompressed before it can be
-     * returned. Hence there is little advantage in allowing subclasses to return a more specific type
-     * as we do for {@link #shallowEntries()}.
-     * @return An iterator over the deep entries of the log
-     */
-    Iterable<LogEntry> deepEntries();
+    Iterable<? extends LogEntry> entries();
```

Comment:
The reference ByteBufferLogInputStream.ByteBufferLogEntry in the comment above is no longer valid.